packages/repository-tests/src/crud/relations/acceptance/has-many-through.relation.acceptance.ts

```diff
@@ -0,0 +1,225 @@
+) {
+  describe('HasManyThrough relation (acceptance)', () => {
+    before(deleteAllModelsInDefaultDataSource);
+    let customerRepo: CustomerRepository;
+    let cartItemRepo: CartItemRepository;
+    let customerCartItemLinkRepo: CustomerCartItemLinkRepository;
+    let existingCustomerId: MixedIdType;
+
+    before(
+      withCrudCtx(async function setupRepository(ctx: CrudTestContext) {
+        ({
+          customerRepo,
+          cartItemRepo,
+          customerCartItemLinkRepo,
+        } = givenBoundCrudRepositories(
+          ctx.dataSource,
+          repositoryClass,
+          features,
+        ));
+        await ctx.dataSource.automigrate([
+          Customer.name,
+          CartItem.name,
+          CustomerCartItemLink.name,
+        ]);
+      }),
+    );
+
+    beforeEach(async () => {
+      await customerRepo.deleteAll();
+      await cartItemRepo.deleteAll();
+      await customerCartItemLinkRepo.deleteAll();
+    });
+
+    beforeEach(async () => {
+      existingCustomerId = (await givenPersistedCustomerInstance()).id;
+    });
+
+    it('can create an instance of the related model alone with a through model', async () => {
+      const item = await customerRepo
+        .cartItems(existingCustomerId)
+        .create(
+          {description: 'an item'},
+          {throughData: {description: 'a through model'}},
+        );
+
+      expect(toJSON(item)).containDeep(
+        toJSON({
+          id: item.id,
+          description: 'an item',
+        }),
+      );
+
+      const persistedItem = await cartItemRepo.findById(item.id);
+      expect(toJSON(persistedItem)).to.deepEqual(toJSON(item));
+      const persistedLink = await customerCartItemLinkRepo.find();
+      expect(toJSON(persistedLink[0])).to.containDeep(
+        toJSON({
+          customerId: existingCustomerId,
+          cartItemId: item.id,
+          description: 'a through model',
+        }),
+      );
+    });
+
+    it('can find instances of the related model', async () => {
+      const item = await customerRepo
+        .cartItems(existingCustomerId)
+        .create(
+          {description: 'an item'},
+          {throughData: {description: 'a through model'}},
+        );
+      const notMyItem = await cartItemRepo.create({
+        description: "someone else's item",
+      });
+      const result = await customerRepo.cartItems(existingCustomerId).find();
+      expect(toJSON(result)).to.containDeep(toJSON([item]));
+      expect(toJSON(result[0])).to.not.containEql(toJSON(notMyItem));
+    });
+
+    it('can patch an instance', async () => {
+      const item1 = await customerRepo
+        .cartItems(existingCustomerId)
+        .create({description: 'group 1'});
+      const item2 = await customerRepo
+        .cartItems(existingCustomerId)
+        .create({description: 'group 1'});
+      const count = await customerRepo
+        .cartItems(existingCustomerId)
+        .patch({description: 'group 2'}, {id: item2.id});
```

Comment:
nitpick: If filter `{id: item2.id}` not provided, would it patch all items belong to customer? If so maybe add one more test for patch all with no filter.